std_execution.bs

```diff
@@ -1247,6 +1248,133 @@ talk about the semantics of some of the algorithms proposed in this paper.
+resolve potential race conditions between the start and completion of the work.
+
+When an algorithm needs to deal with potentially eager senders, the potential
+race conditions can be resolved one of several ways, none of which is desirable:
+
+1.  Assume the worst and implement the algorithm defensively, assuming all
+    senders are eager. This obviously has overheads both at runtime and in
+    algorithm complexity. Resolving race conditions is hard.
+
+2.  Require senders to declare whether they are eager or not with a query.
+    Algorithms can then implement two different implementation strategies, one
+    for strictly lazy senders and one for potentially eager senders. This
+    addresses the performance problem of (1) while compounding the complexity
+    problem.
+
+3.  Push the problem back onto eager senders by requiring them to never call a
+    receiver's `set_*` completion until after `execution::start` has been called
+    on the operation state. That puts the onus of dealing with the race
+    condition on the sender author, which is arguably where it belongs. But it
+    has the negative effect of delaying the completion of the operation. Even
+    when the work is done and the continuation available, the sender has to
+    withold the completion, possibly indefinitely.
+
+Of these options, (3) seems the least bad but would seem to run contrary to the
+<i>raison d'être</i> of eager algorithms, which ostensibly is to make the
+results available sooner.
+
+### Eager senders incur cancellation-related overhead ### {#design-lazy-algorithms-runtime}
+
+Another implication of the use of eager operations is with regards to
+cancellation. The eagerly executing operation will not have access to the
+caller's stop token until the sender is connected to a receiver. If we still
+want to be able to cancel the eager operation then it will need to create a new
+stop source and pass its associated stop token down to child operations. Then
+when the returned sender is eventually connected it will register a stop
+callback with the receiver's stop token that will request stop on the eager
+sender's stop source.
+
+As the eager operation does not know at the time that it is launched what the
+type of the receiver is going to be, and thus whether or not the stop token
+returned from `execution::get_stop_token` is an `std::unstoppable_token` or not,
+the eager operation is going to need to assume it might be later connected to a
+receiver with a stop token that might actually issue a stop request. Thus it
+needs to incur the runtime overhead of supporting cancellation, even if
+cancellation will never be requested by the caller.
+
+The eager operation will also need to do this to support sending a stop request
+to the eager operation in the case that the sender representing the eager work
+is destroyed before it has been joined (assuming strategy (5) or (6) listed
+above is chosen).
+
```

Comment:
Other topics you didn't mention here:
* Extra synchronisation/atomic-ops required to resolve race between concurrent connect/start and completion of the operation.
* Extra heap-allocation required to hold state of the operation-state of the eager operation while it's executing before the returned sender has been connected/started.
* Lack of the ability to implicitly propagate context from the receiver to the eager operation. As the operation may run to completion before a receiver is connected/started, we won't necessarily have a receiver available which can obtain implicitly propagated context information from (like the current scheduler, stop token, priority, deadline, etc.).